docs/best-practices/continuous-integration.md

```diff
@@ -18,49 +18,49 @@ Multiple CI products and services offer integrations with fastlane:
 
-There is nothing _fastlane_ can do better in that regard, as these are technical limitations on how App Store Connect sessions are handled.
+The easiest way to get _fastlane_ running on a CI system is to create a separate Apple ID that doesn't have 2-factor auth enabled, with a long, randomly generated password. Additionally make sure the newly created Apple account has limited access to only the apps and resources it needs.
 
 Creating a separate Apple ID allows you to limit the permission scope, have a randomly generated password, and will make it much easier for you to set up CI using _fastlane_.
 
 ### Security code and session
 
-When your Apple account has 2-factor or 2-step auth enabled, you'll be asked to verify your identity by entering a security code. If you already have a trusted device configured for your account, then the code will appear on the device. If you don't have any devices configured, but have trusted a phone number, then the code will be sent to your phone. The resulting session will be stored in `~/.fastlane/spaceship/[email]/cookie`. Again, the session should be valid for about one month, however there is no way to test this without actually waiting for over a month.
+When your Apple account has 2 Factor Authentication (or 2 step verification) enabled, you will be asked to verify your identity by entering a security code. If you already have a trusted device configured for your account, then the code will appear on the device. If you don't have any devices configured, but have trusted a phone number, then the code will be sent to your phone. The resulting session will be stored in `~/.fastlane/spaceship/[email]/cookie`. The session should be valid for about one month, however there is no way to test this without actually waiting for over a month.
 
 ### Use of application specific passwords and `spaceauth`
 
-Before going through this guide, make sure to read the section above.
+When you can not enter this token, as on a Continuous Integration system, you have to use other ways to log in:
+
+#### Application specific passwords
 
 If you want to upload builds to TestFlight / App Store Connect from your CI machine, you need to generate an application specific password:
 
 1. Visit [appleid.apple.com/account/manage](https://appleid.apple.com/account/manage)
 1. Generate a new application specific password
 1. Provide the application specific password using the environment variable `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`
 
-Because your CI machine will not be able to prompt you for your two-step or two-factor auth information, you also need to generate a login session for your CI machine in advance. You can get this by running:
+#### `spaceauth`
+
+Because your CI machine will not be able to prompt you for your two-step or two-factor auth information, you also need to generate a login session for your CI machine in advance. You can get on your local machine this by running:
```

Comment:
```suggestion
Because your CI machine will not be able to prompt you for your 2-step or 2-factor auth information, you also need to generate a login session for your CI machine in advance. You can get this on your local machine by running:
```